src/debug_mode_test.ts

```diff
@@ -35,10 +35,9 @@ describeWithFlags('debug on', ALL_ENVS, () => {
     expectArraysClose(res, [2, 0, 0, 3]);
   });
 
-  it('debug mode errors when there are nans, float32', () => {
-    const a = tf.tensor1d([2, NaN]);
-    const f = () => tf.relu(a);
-    expect(f).toThrowError();
+  it('debug mode errors when nans in tensor construction, float32', () => {
+    const a = () => tf.tensor1d([2, NaN], 'float32');
+    expect(a).toThrowError();
   });
```

Comment:
add a unit test making sure that even though the user didn't upload any NaNs, but a kernel produced those, it still errors. Another similar test for Infinities. For example do tf.div(a, b) where b has 0. Likewise tf.log(a) where a has zeros

Reply:
Done